samples/net/sockets/http_server/src/server.c

```diff
@@ -0,0 +1,979 @@
+
+struct http_parser_settings parserSettings;
+struct http_parser parser;
+
+static unsigned char settings_frame[9] = {0x00, 0x00, 0x00, /* Length */
+	0x04, /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x00, /* Flags: 0x00 - unused flags */
+	0x00, 0x00, 0x00,
+	0x00}; /* Reserved, Stream Identifier: 0x00 - overall connection */
+
+static unsigned char settings_ack[9] = {0x00, 0x00, 0x00, /* Length */
+	0x04, /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x01, /* Flags: 0x01 - ACK */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier */
+
+static struct http2_frame frame;
+
+const char content[] = {
+#include "index.html.gz.inc"
+};
+
+bool has_upgrade_header = true;
+
+int http2_server_init(
+	struct http2_server_ctx *ctx, struct http2_server_config *config)
+{
+	/* Create a socket */
+	ctx->server_fd = socket(config->address_family, SOCK_STREAM, 0);
+	if (ctx->server_fd < 0) {
+		LOG_ERR("socket");
+		return ctx->server_fd;
+	}
+
+	if (setsockopt(ctx->server_fd, SOL_SOCKET, SO_REUSEADDR, &(int){1},
+		    sizeof(int)) < 0) {
+		LOG_ERR("setsockopt");
+		return -errno;
+	}
+
+	/* Set up the server address struct according to address family */
+	if (config->address_family == AF_INET) {
+		struct sockaddr_in serv_addr;
+
+		memset(&serv_addr, 0, sizeof(serv_addr));
+		serv_addr.sin_family = AF_INET;
+		serv_addr.sin_addr.s_addr = INADDR_ANY;
+		serv_addr.sin_port = htons(config->port);
+
+		if (bind(ctx->server_fd, (struct sockaddr *)&serv_addr,
+			    sizeof(serv_addr)) < 0) {
+			LOG_ERR("bind");
+			return -errno;
+		}
+	} else if (config->address_family == AF_INET6) {
+		struct sockaddr_in6 serv_addr;
+
+		memset(&serv_addr, 0, sizeof(serv_addr));
+		serv_addr.sin6_family = AF_INET6;
+		serv_addr.sin6_addr = in6addr_any;
+		serv_addr.sin6_port = htons(config->port);
+
+		if (bind(ctx->server_fd, (struct sockaddr *)&serv_addr,
+			    sizeof(serv_addr)) < 0) {
+			LOG_ERR("bind");
+			return -errno;
+		}
+	}
+
+	/* Listen for connections */
+	if (listen(ctx->server_fd, MAX_CLIENTS) < 0) {
+		LOG_ERR("listen");
+		return -errno;
+	}
+
+	/* Create an eventfd*/
+	ctx->event_fd = eventfd(0, 0);
+	if (ctx->event_fd < 0) {
+		LOG_ERR("eventfd");
+		return -errno;
+	}
+
+	/* Initialize fds */
+	memset(ctx->fds, 0, sizeof(ctx->fds));
+	memset(ctx->clients, 0, sizeof(ctx->clients));
+
+	ctx->fds[0].fd = ctx->server_fd;
+	ctx->fds[0].events = POLLIN;
+
+	ctx->fds[1].fd = ctx->event_fd;
+	ctx->fds[1].events = POLLIN;
+
+	ctx->num_clients = 0;
+	ctx->infinite = 1;
+
+	return ctx->server_fd;
+}
+
+int accept_new_client(int server_fd)
+{
+	int new_socket;
+
+	socklen_t addrlen;
+	struct sockaddr_in sa;
+
+	memset(&sa, 0, sizeof(sa));
+	addrlen = sizeof(sa);
+
+	new_socket = accept(server_fd, (struct sockaddr *)&sa, &addrlen);
+	if (new_socket < 0) {
+		LOG_ERR("accept failed");
+		return new_socket;
+	}
+
+	return new_socket;
+}
+
+int http2_server_start(struct http2_server_ctx *ctx)
```

Comment:
so again, we are doing more than http2, so please replace `http2` with `http` everywhere, unless it's a header specifically describing http2-only protocol bits.